### PR TITLE
Handle more than one default gateway in debug script

### PR DESCRIPTION
**Pull request by yubiuser**
**What does this PR aim to accomplish?:**

Fixes https://github.com/pi-hole/pi-hole/issues/4933

If more than one default gateway is associated with an interface (usually IPv6) we saved all of them as a single string in `gateway` and then try to ping this during the debug run. Ping this "address" failed.

Before:
```
[i] Default IPv4 gateway: 10.0.1.1
   * Pinging 10.0.1.1...
[✓] Gateway responded.
[i] Default IPv6 gateway: fe80::464e:6dff:fecf:5cd8
fe80::464e:6dff:fecf:5cd8
   * Pinging fe80::464e:6dff:fecf:5cd8
fe80::464e:6dff:fecf:5cd8...
ping: fe80::464e:6dff:fecf:5cd8
fe80::464e:6dff:fecf:5cd8: Name or service not known
[✗] Gateway did not respond.
```
  
After:
```
[i] Default IPv4 gateway(s):
     10.0.1.1
   * Pinging first gateway 10.0.1.1...
[✓] Gateway responded.
[i] Default IPv6 gateway(s):
     fe80::464e:6dff:fecf:5cd8
     fe80::464e:6dff:fecf:5cd8
   * Pinging first gateway fe80::464e:6dff:fecf:5cd8...
ping6: Warning: source address might be selected on device other than eth0.
[✓] Gateway responded.
```

- **How does this PR accomplish the above?:**

Create an array containing all the default gateways for one interface and only try to ping the first one.

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
- [x] I have read the above and my PR is ready for review. _Check this box to confirm_

### CODE REVIEW

**Comment by dschaper** [comment]
Do you think we really need to ping6 addresses? I like the changes for IPv4 but I'm wondering if there is any use to ping IPv6 addresses since they could be LL that don't tell us much and trying to decide what type of IPv6 addresses they are is way to deep in the weeds.

File: advanced/Scripts/piholeDebug.sh [hunk 1]
```diff
@@ -680,17 +680,18 @@ ping_gateway() {
     # Check if we are using IPv4 or IPv6
     # Find the default gateway using IPv4 or IPv6
     local gateway
-    gateway="$(ip -"${protocol}" route | grep default | grep "${PIHOLE_INTERFACE}" | cut -d ' ' -f 3)"
+    mapfile -t gateway < <(ip -"${protocol}" route | grep default | grep "${PIHOLE_INTERFACE}" | cut -d ' ' -f 3)
```
Comment (dschaper) [inline advanced/Scripts/piholeDebug.sh:683]: `mapfile` isn't POSIX and isn't supported in `dash`.

File: advanced/Scripts/piholeDebug.sh [hunk 2]
```diff
@@ -680,17 +680,18 @@ ping_gateway() {
...
 
-    # If the gateway variable has a value (meaning a gateway was found),
-    if [[ -n "${gateway}" ]]; then
-        log_write "${INFO} Default IPv${protocol} gateway: ${gateway}"
+    # If the gateway array is not empty (meaning a gateway was found),
+    if [[ ${#gateway[@]} -ne 0 ]]; then
```
Comment (dschaper) [inline advanced/Scripts/piholeDebug.sh:686]: Same issue here with bashisms.

**Review by dschaper — CHANGES_REQUESTED** [review]

**Comment by yubiuser** [comment]
Users might be on IPv6-only networks. We could check if the device has a GUA address and ping the respective gateway

File: advanced/Scripts/piholeDebug.sh [hunk 3]
```diff
@@ -678,20 +678,24 @@ ping_gateway() {
...
+    # If there was at least one gateway
+    if [ -n "${gateway}" ]; then
         # Let the user know we will ping the gateway for a response
-        log_write "   * Pinging first gateway ${gateway[0]}..."
+        log_write "   * Pinging first gateway ${gateway}..."
         # Try to quietly ping the gateway 3 times, with a timeout of 3 seconds, using numeric output only,
         # on the pihole interface, and tail the last three lines of the output
-        # If pinging the gateway is not successful,
-        if ! ${cmd} -c 1 -W 2 -n "${gateway[0]}" -I "${PIHOLE_INTERFACE}" >/dev/null; then
+        # If pinging" the gateway is not successful,
```
Comment (rdwebdesign) [inline advanced/Scripts/piholeDebug.sh:697]: I think this `"` quote was a typo.

**Comment by dschaper** [comment]
> We could check if the device has a GUA address and ping the respective gateway

That's the kind of complexity that I'd like to avoid. I can see the rare case that a network may be IPv6 only, I don't know how useful the ping test is in that case (and really in any case) since you need a gateway of some type in order to get the debug log on to `tricorder` in the first place.

**Review by dschaper — APPROVED** [review]

**Pull request merged**